trace-obfuscation/src/replacer.rs

```diff
@@ -0,0 +1,198 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use crate::pb;
+use regex::Regex;
+
+pub trait TraceTagReplacer {
```

Comment:
program the trace tag replacer to a trait, so that users of the agentless code can swap out and implement their own tag replacer, based on their scenario